hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/local/LocalRegionCache.java

```diff
@@ -195,18 +193,28 @@ protected Object createMessage(final Object key, final Object value, final Objec
     protected MessageListener<Object> createMessageListener() {
         return new MessageListener<Object>() {
             public void onMessage(final Message<Object> message) {
+                if (message.getPublishingMember().localMember()) {
+                    return;
+                }
                 final Invalidation invalidation = (Invalidation) message.getMessageObject();
```

Comment:
I'd probably gap these apart, after your `return` above. I might even split all this invalidation logic into a `maybeInvalidate(final Message<Object> message)` method, just to enhance the readability of the code